### ISSUE TITLE
diameter Address type only handles IP-addresses

### ISSUE THREAD

**Issue by sebastiw** [erlang/otp#5463]
**Describe the bug**

Diameter crashes when receiving an E.164 address in an AVP of type `Address`.

**To Reproduce**
E.g. Receiving an [Ro/CCR](https://www.etsi.org/deliver/etsi_ts/132200_132299/132299/16.02.00_60/ts_132299v160200p.pdf) with Client-Address set to an E.164 address.

**Expected behavior**
No crash, but return the octet string/decoded address.

**Affected versions**
All OTP versions since R14 (since the introduction of the diameter application)

**Additional context**

According to [RFC3588 section 4.3](https://datatracker.ietf.org/doc/html/rfc3588#section-4.3) the Address type is of OctetString where the first octet represents the Address Type.

> The first two octets of the Address AVP represents the Address
> Type, which contains an Address Family defined in [IANAADFAM]. 
> The AddressType is used to discriminate the content and format of the remaining octets.

And according to [IANAADFAM](https://www.iana.org/assignments/address-family-numbers/address-family-numbers.xhtml) there is a lot of different Address Types, but only IPv4 and IPv6 are handled by diameter application.

From [diameter_types.erl](https://github.com/erlang/otp/blob/master/lib/diameter/src/base/diameter_types.erl#L245) only IP-addresses are allowed.

```
'Address'(encode, zero, _) ->
    <<0:48>>;

'Address'(decode, <<A:16, B/binary>>, _)
  when 1 == A,  4 == size(B);
       2 == A, 16 == size(B) ->
    list_to_tuple([N || <<N:A/unit:8>> <= B]);

'Address'(decode, B, _) ->
    ?INVALID_LENGTH(B);

'Address'(encode, T, _) ->
    Ns = tuple_to_list(diameter_lib:ipaddr(T)),  %% length 4 or 8
    A = length(Ns) div 4,                        %% 1 or 2
    B = << <<N:A/unit:8>> || N <- Ns >>,
    <<A:16, B/binary>>.
```

Best regards

**Comment by otp-anders** [comment]
You're right that support for other address types is missing (which I'll fix), but where is the crash? Reception of It should just result in a decode error that's handled like any other.

**Comment by sebastiw** [comment]
sorry it was most likely a decode error (I got the issue from a colleague)

**Comment by otp-anders** [comment]
Addressed in 25.0. (Missed closing this.)
